File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Entities/UserConversation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Entities
{
    public class UserConversation : Entity
    {
        [Key]
        public Guid UserConversationId { get; set; } = Guid.Empty;
        public Guid UserId { get; set; } = Guid.Empty;
        public Guid ConversationId { get; set; } = Guid.Empty;
        public string? Nickname { get; set; }
        public DateTime Time { get; set; }
        public DateTime LastReadTime { get; set; }
        public DateTime LastDeleteTime { get; set; }
        protected override UserConversation Init()
        {
            return new UserConversation();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/user-conversation.js

import Entity from './entity.js';

class UserConversation extends Entity{
    constructor() {
        super();
        this.UserConversationId = null;
        this.UserId = null;
        this.ConversationId = null;
        this.Nickname = null;
        this.Time = null;
        this.LastReadTime = null;
        this.LastDeleteTime = null;
    }

    init() {
        return new UserConversation();
    }
}

export default UserConversation;

